### ISSUE TITLE
Possibility to add custom headers for HTTP downloads

### ISSUE THREAD

**Issue by dburgermeister** [carvel-dev/vendir#87]
**Describe the problem/challenge you have**
Some websites request to add specific HTTP headers, like for example API Token to access the files.
Currently it seems not possible to add those headers in the yaml file.
Example: wget -O <filename> --header="Authorization: Token *************" <url>

**Describe the solution you'd like**
Add a new field in the vendir.yaml configuration file for each header.

**Anything else you would like to add:**
Thank you :)

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by cppforlife** [comment]
@dburgermeister yeah that makes sense. we probably would want to pull them in from a secret since sounds like some headers might include sensitive information. would you be interested in contributing this?

**Comment by voor** [comment]
This use case came up again recently with someone adopting HTTP fetch in the App CR.

They specifically used the GitHub API, and were unable to add authorization to the URL: https://docs.github.com/en/rest/overview/authenticating-to-the-rest-api?apiVersion=2022-11-28
